pkg/conversion/capi2mapi/aws.go

```diff
@@ -167,6 +168,25 @@ func (m machineAndAWSMachineAndAWSCluster) toProviderSpec() (*mapiv1beta1.AWSMac
 		MarketType:              mapiAWSMarketType,
 	}
 
+	// Dedicated host support
+	if m.awsMachine.Spec.HostAffinity != nil {
+		switch *m.awsMachine.Spec.HostAffinity {
+		case "host":
+			mapaProviderConfig.HostPlacement = &mapiv1beta1.HostPlacement{
+				Affinity: ptr.To(mapiv1beta1.HostAffinityDedicatedHost),
+				DedicatedHost: &mapiv1beta1.DedicatedHost{
+					ID: *m.awsMachine.Spec.HostID,
```

Comment:
do we need to check `HostID` isnt nil to be safe or is there already a previous validation of this scenario?